python/pyspark/sql/tests/test_udtf.py

```diff
@@ -3437,6 +3437,52 @@ def eval(self):
                     udtf(TestUDTF, returnType=ret_type)().collect()
 
 
+def test_udtf_with_collated_string_types(self):
+    class MyUDTF:
```

Comment:
For UDTFs, can we try using the DDL string format as well? Like `@udtf("out1 string collate ..., out2 string ...")